packages/substrate-bindings/src/codecs/scale/Binary.ts

```diff
@@ -54,7 +54,10 @@ const enc = (nBytes?: number): Encoder<Binary> => {
 
 const dec = (nBytes?: number): Decoder<Binary> => {
   const _dec = Bytes.dec(nBytes)
-  return (value) => Binary.fromBytes(_dec(value))
+  return (value) =>
+    nBytes === undefined
+      ? Binary.fromBytes(_dec(value))
+      : FixedSizeBinary.fromBytes(_dec(value))
```

Comment:
small nit:

```suggestion
  const Bin = nBytes == null ? Binary : FixedSizeBinary
  return (value) => Bin.fromBytes(_dec(value))
```

Reply:
FWIW: the `== null` is a trick used in libraries (which I wouldn't use when working on a top-level App) that produces `true` if the value is `undefined` or `null` and `false` otherwise. It has the advantage that it produces slightly smaller bundle sizes 🙂.